airflow/contrib/hooks/gcp_dlp_hook.py

```diff
@@ -0,0 +1,1362 @@
+            in the template.
+        :type deidentify_template_name: str
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        :rtype: google.cloud.dlp_v2.types.DeidentifyContentResponse
+        """
+
+        client = self.get_conn()
+
+        if not parent:
+            raise AirflowException("Please provide the parent resource name.")
+
+        return client.deidentify_content(
+            parent=parent,
+            deidentify_config=deidentify_config,
+            inspect_config=inspect_config,
+            item=item,
+            inspect_template_name=inspect_template_name,
+            deidentify_template_name=deidentify_template_name,
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def delete_deidentify_template(
+        self, name, retry=None, timeout=None, metadata=None
+    ):
+        """
+        Deletes a DeidentifyTemplate.
+
+        :param name: Resource name of the organization and deidentify template to be deleted.
+        :type name: str
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: sequence[tuple[str, str]]]
+        """
+
+        client = self.get_conn()
+
+        if not name:
+            raise AirflowException(
+                "Please provide the deidentifyTemplate name to be deleted."
+            )
+
+        client.delete_deidentify_template(
+            name=name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def delete_dlp_job(
+        self, name, retry=None, timeout=None, metadata=None
+    ):
+        """
+        Deletes a long-running DlpJob. This method indicates that the client is no longer
+        interested in the DlpJob result. The job will be cancelled if possible.
+
+        :param name: The name of the DlpJob resource to be deleted.
```

Comment:
It is worth adding the possibility of defining the project ID using the connection configuration.